### PR TITLE
[#9493] Convert FieldValidator to a utility class

### LINKED ISSUE #9493: Convert FieldValidator to a utility class
The `FieldValidator` class right now is implemented as a standard initialize-able class that can supposedly hold data. However, it is used as a utility class and contains no internal data that is being mutated.

Therefore it should be designed as a utility class, i.e. all methods are static, the class is `final`, and the constructor is `private`.

### PR DESCRIPTION
**Pull request by VinodKumarWagh**
Fixes #9493
Convert all files according to the functionality of the new FieldValidator utility class.

### CODE REVIEW

File: src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java [hunk 1]
```diff
@@ -198,55 +198,56 @@ public String getJsonString() {
...
     @Override
     public List<String> getInvalidityInfo() {
-        FieldValidator validator = new FieldValidator();
         List<String> errors = new ArrayList<>();
 
         // Check for null fields.
 
-        addNonEmptyError(validator.getValidityInfoForNonNullField(
+        addNonEmptyError(
+                FieldValidator.getValidityInfoForNonNullField(
```
Comment (amrut-prabhu) [inline src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java]: Umm let's move this to the previous line as well. Similar to line 209

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 2]
```diff
@@ -267,6 +267,11 @@
             Arrays.asList(
                     new String[] {"comments?", "notes?"}));
 
+    // Intentional private constructor to prevent instantiation.
```
Comment (amrut-prabhu) [inline src/main/java/teammates/common/util/FieldValidator.java]: Move this to the line after `// utility class`

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 3]
```diff
@@ -267,6 +267,11 @@
...
+    private FieldValidator() {
+      // utility class
```
Comment (amrut-prabhu) [inline src/main/java/teammates/common/util/FieldValidator.java]: We use 4-space indentation for Java

File: src/main/java/teammates/ui/template/InstructorFeedbackResultsGroupByQuestionPanel.java [hunk 4]
```diff
@@ -20,7 +20,7 @@ public InstructorFeedbackResultsGroupByQuestionPanel(List<InstructorFeedbackResu
         this.name = participantName;
         this.isGiver = isGiver;
 
-        this.isEmailValid = new FieldValidator()
+        this.isEmailValid = FieldValidator
```
Comment (amrut-prabhu) [inline src/main/java/teammates/ui/template/InstructorFeedbackResultsGroupByQuestionPanel.java]: We can move the next 2 lines up here as well

File: src/main/java/teammates/ui/webapi/request/AccountCreateRequest.java [hunk 5]
```diff
@@ -41,24 +41,25 @@ public void setInstructorEmail(String instructorEmail) {
...
 
-        String nameError = validator.getInvalidityInfoForPersonName(this.instructorName.trim());
+        String nameError = FieldValidator.getInvalidityInfoForPersonName(this.instructorName.trim());
         if (!nameError.isEmpty()) {
             errors.add(nameError);
         }
 
-        String emailError = validator.getInvalidityInfoForEmail(this.instructorEmail.trim());
+        String emailError = FieldValidator.getInvalidityInfoForEmail(this.instructorEmail.trim());
+
```
Comment (amrut-prabhu) [inline src/main/java/teammates/ui/webapi/request/AccountCreateRequest.java]: Why the additional newline?

File: src/main/java/teammates/ui/webapi/request/AccountCreateRequest.java [hunk 6]
```diff
@@ -41,24 +41,25 @@ public void setInstructorEmail(String instructorEmail) {
...
         if (!emailError.isEmpty()) {
             errors.add(emailError);
         }
 
-        String instituteError = validator.getInvalidityInfoForInstituteName(this.instructorInstitution.trim());
+        String instituteError = FieldValidator.getInvalidityInfoForInstituteName(this.instructorInstitution.trim());
+
```
Comment (amrut-prabhu) [inline src/main/java/teammates/ui/webapi/request/AccountCreateRequest.java]: same here

File: src/test/java/teammates/test/cases/util/FieldValidatorTest.java [hunk 7]
```diff
@@ -216,8 +210,9 @@ public void testGetInvalidityInfoForPersonName_invalid_returnSpecificErrorString
 
     @Test
     public void testGetInvalidityInfoForInstituteName_invalid_returnSpecificErrorString() {
-        String invalidInstituteName = StringHelperExtension.generateStringOfLength(INSTITUTE_NAME_MAX_LENGTH + 1);
-        String actual = validator.getInvalidityInfoForInstituteName(invalidInstituteName);
+        String invalidInstituteName = StringHelperExtension.generateStringOfLength(
+                        FieldValidator.INSTITUTE_NAME_MAX_LENGTH + 1);
```
Comment (amrut-prabhu) [inline src/test/java/teammates/test/cases/util/FieldValidatorTest.java]:
```suggestion
                                                                FieldValidator.INSTITUTE_NAME_MAX_LENGTH + 1);
```

File: src/test/java/teammates/test/cases/util/FieldValidatorTest.java [hunk 8]
```diff
@@ -228,27 +223,28 @@ public void testGetInvalidityInfoForInstituteName_invalid_returnSpecificErrorStr
...
         assertEquals("Invalid nationality (invalid char) should return error string that is specific to nationality",
-                     String.format(NATIONALITY_ERROR_MESSAGE, invalidNationality), actual);
+                     String.format(FieldValidator.NATIONALITY_ERROR_MESSAGE, invalidNationality), actual);
 
         invalidNationality = "<script> alert('hi!'); </script>";
-        actual = validator.getInvalidityInfoForNationality(invalidNationality);
+        actual = FieldValidator.getInvalidityInfoForNationality(invalidNationality);
         assertEquals("Unsanitized, invalid nationality should return sanitized error string",
-                String.format(NATIONALITY_ERROR_MESSAGE, SanitizationHelper.sanitizeForHtml(invalidNationality)), actual);
+                String.format(
```
Comment (amrut-prabhu) [inline src/test/java/teammates/test/cases/util/FieldValidatorTest.java]: I don't think you need to move the rest to a new line

File: src/test/java/teammates/test/cases/util/FieldValidatorTest.java [hunk 9]
```diff
@@ -279,8 +275,9 @@ public void testGetInvalidityInfoForCourseName_invalid_returnSpecificErrorString
 
     @Test
     public void testGetInvalidityInfoForFeedbackSessionName_invalid_returnSpecificErrorString() {
-        String invalidSessionName = StringHelperExtension.generateStringOfLength(FEEDBACK_SESSION_NAME_MAX_LENGTH + 1);
-        String actual = validator.getInvalidityInfoForFeedbackSessionName(invalidSessionName);
+        String invalidSessionName = StringHelperExtension.generateStringOfLength(
+                FieldValidator.FEEDBACK_SESSION_NAME_MAX_LENGTH + 1);
```
Comment (amrut-prabhu) [inline src/test/java/teammates/test/cases/util/FieldValidatorTest.java]: Same as earlier- add indentation

File: src/test/java/teammates/test/cases/util/FieldValidatorTest.java [hunk 10]
```diff
@@ -527,64 +529,67 @@ public void testGetInvalidityInfoForCourseId_invalid_returnErrorString() {
...
         assertEquals("Invalid Course ID (whitespace only) should return appropriate error string",
-                     WHITESPACE_ONLY_OR_EXTRA_WHITESPACE_ERROR_MESSAGE.replace("${fieldName}", COURSE_NAME_FIELD_NAME),
-                     validator.getInvalidityInfoForCourseId(whitespaceOnlyCourseId));
+                     FieldValidator.WHITESPACE_ONLY_OR_EXTRA_WHITESPACE_ERROR_MESSAGE.replace(
+                        "${fieldName}", FieldValidator.COURSE_NAME_FIELD_NAME),
+                     FieldValidator.getInvalidityInfoForCourseId(whitespaceOnlyCourseId));
 
-        String tooLongCourseId = StringHelperExtension.generateStringOfLength(COURSE_ID_MAX_LENGTH + 1);
+        String tooLongCourseId = StringHelperExtension.generateStringOfLength(
+                FieldValidator.COURSE_ID_MAX_LENGTH + 1);
```
Comment (amrut-prabhu) [inline src/test/java/teammates/test/cases/util/FieldValidatorTest.java]: Add indentation (like earlier) to improve readability

File: src/test/java/teammates/test/cases/webapi/UpdateInstructorActionTest.java [hunk 11]
```diff
@@ -83,7 +82,9 @@ protected void testExecute() {
         assertEquals(HttpStatus.SC_BAD_REQUEST, actionOutput.getStatusCode());
 
         msg = (MessageOutput) actionOutput.getOutput();
-        String expectedErrorMessage = FIELD_VALIDATOR.getInvalidityInfoForEmail(invalidEmail);
+
```
Comment (amrut-prabhu) [inline src/test/java/teammates/test/cases/webapi/UpdateInstructorActionTest.java]: Remove these 2 unnecessary newlines

Comment (VinodKumarWagh) [reply src/test/java/teammates/test/cases/webapi/UpdateInstructorActionTest.java]: @amrut-prabhu Thanks for the review. Apologies of the mistakes, I have corrected all the reviewed corrections.

**Comment by VinodKumarWagh** [comment]
@amrut-prabhu is there any reviewed correction still left? I had given my best in case of corrections! some of them were not corrected because of the lint errors, but i adjusted them somehow to increase the readablity of the code.

File: src/main/java/teammates/common/util/FieldValidator.java [hunk 12]
```diff
@@ -267,6 +267,11 @@
             Arrays.asList(
                     new String[] {"comments?", "notes?"}));
 
+    private FieldValidator() {
+    // utility class
```
Comment (amrut-prabhu) [inline src/main/java/teammates/common/util/FieldValidator.java]: Add one more level of indentation to both lines (so, 4 spaces).  Like how it's done in `Assumption.java`

**Comment by VinodKumarWagh** [comment]
@wkurniawan07 @amrut-prabhu @RonakLakhotia @rrtheonlyone @monmanuela Thank you all for helping and guiding me. This was my very first contribution to the world of open source. I had a great experience on this platform and it has taught me a lot of valuable lessons, and I don't think I could ever do that task without you all. This experience has motivated me to contribute something more valuable stuff in the future. :)

**Pull request merged**